notifications-yaml-reference_5be45bb02c7d3a31944dbc98.md

```diff
@@ -0,0 +1,207 @@
+multiple notifications, you will need to have multiple `rules` items.
+
+## The rules property in more detail
+
+In this section we are going to explain the properties that are stored in a
+`rules` property – the logic of each notification is stored under a `rules`
+property.
+
+### name
+
+The value of the `name` property, which is a string, in the `metadata` context
+defines the name of the rule of the notification.
+
+### filter
+
+The `filter` property holds the `projects`, `branches` and `pipelines`
+properties.
+
+The values of `--branches`, `--projects` and `--pipelines` can contain regular
+expressions. Regex matches must be wrapped in forward slashes (`/.*/`).
+Specifying a branch name without slashes (example: `.*`) would execute a direct
+equality match.
+
+For a filter to be `true` all of its conditions should be true. If a condition
+such as `branches` has multiple cases, only one of these cases should be a
+match.
+
+#### projects
+
+The `projects` property is used for holding a list of Semaphore 2.0 projects
+that interest us.
+
+#### pipelines
+
+The `pipelines` property is used for holding a list of pipeline filenames that
+interest us.
+
+#### branches
+
+The `branches` property holds a list of strings, which should be GitHub branch
+names, that interest us.
+
+### notify
+
+The `notify` property holds the `slack` property that is used for defining
+information related to the Slack channel or channels related to the defined
+notification.
+
+#### slack
+
+The `slack` property holds information related to Slack such as the URL of the
+Incoming WebHook, the message that will be displayed, etc.
+
+##### endpoint
+
+The `endpoint` property holds the URL of the Incoming WebHook that will be used.
+Incoming WebHooks offer a simple way for posting messages from external sources
+into Slack.
+
+You can learn more about defining the value of the `endpoint` property by
+visiting https://slack.com/apps/A0F7XDUAZ-incoming-webhooks.
+
+Please note that if a notification does not work as expected, you might want to
+begin the debugging process by verifying the Slack WebHook used.
+
+Tip: you can use just a single Incoming WebHook from Slack for all your
+notifications as this Incoming WebHook has access to all the channels of a
+Slack Workspace.
+
+##### message
+
+The `message` property allows you to define the message that will be displayed
+on the Slack channel or the Slack user by the rule of the notification.
+
+##### channels
+
+The `channels` property holds a list of strings. Each string item is the name
+of a Slack channel or the name of a Slack user.
+
+A notification can send a message to multiple Slack channels and multiple users.
+
+## An example
+
+The following YAML code present an example of a notification as returned by the
+`sem get notification [name]` command:
+
+     sem get notifs docs
+    apiVersion: v1alpha
+    kind: Notification
+    metadata:
+      name: docs
```

Comment:
If this notification is sending notifications for other projects appart from the docs project, I would suggest changing this name to something that is demonstrating the value of describing multiple independent rules in one resource.

For example, `documentation-release-cycle` would be a good notification name examples for a notification resource that describes various aspects of the documentation projects.

However, I would push it in the direction of using common scenarios for software release. Documentation is not the most common use case of notifications. Notifications for releasing and notifying multiples team for new API versions is much more common and instantly recognizable.

Reply:
That is true but the first example should be simple. I renamed it to `API-docs`.